### PR TITLE
Expose BatchNorm flag for UNet

### LINKED ISSUE #471: Exposing Batch norm flag
## Problem

There may be issues with BatchNorm layers when there are many background patches. Let's investigate, but we may benefit from allowing BatchNorm to be disabled in the UNet.

### PR DESCRIPTION
**Pull request by Callidior**
on-behalf-of: @zeiss <bjoern.barz@zeiss.com>

## Description

> [!NOTE]  
> **tldr**:  Enables disabling the use of BatchNorm in U-Net from the user-facing API.

### Background - why do we need this PR?

BatchNorm can cause problems and artifacts, especially when images contain large homogenous regions. It is also rarely necessary for models as shallow as typically used for denoising.

### Overview - what changed?

The flag `use_batch_norm` was added to `UNetModel`.

### Implementation - how did you implement the changes?

Users can now disable BatchNorm via `model_params`:

```python
config = create_n2v_configuration(
    ...
    model_params={"use_batch_norm": False},
)
```

### Modified features or files

- Changed `UNetModel` in `src/careamics/config/architectures/unet_model.py`

## How has this been tested?

- Running an N2V training with BatchNorm disabled works fine

## Related Issues

- Resolves #471 

## Additional Notes and Examples

---

**Please ensure your PR meets the following requirements:**

- [x] Code builds and passes tests locally, including doctests
- [ ] New tests have been added (for bug fixes/features)
- [x] Pre-commit passes
- [ ] PR to the documentation exists (for bug fixes / features)

### CODE REVIEW

**Comment by Callidior** [comment]
@jdeschamps Is this sufficient to resolve #471? It does the job, at least.

**Review by jdeschamps — APPROVED** [review]

**Review by jdeschamps — APPROVED** [review]
Great, thanks! 

I have to check whether there are some tests that need to have the batch norm added for completeness, then we can merge it!

**Comment by jdeschamps** [comment]
No issue, I will merge! Thanks for the contribution! 🎉
